### PR TITLE
Developer environments

### PR DESCRIPTION
**Pull request by AndreaGiardini**

### CODE REVIEW

File: .github/workflows/dagster.yml [hunk 1]
```diff
@@ -29,8 +29,36 @@ on:
         description: 'GCR service account credentials to push/pull Docker images'
         required: true
 
+env:
+  AUTH_URL: auth.stage-elm.overstory.ai
+
 jobs:
+  cleanup-pr:
+    if: github.event.action == 'closed'
```
Comment (chanange) [inline .github/workflows/dagster.yml:37]: Should this also check for the 'unlabelled' action and run conditionally if `dev-env` label exists or is removed?
is there just an error message when helm tries to remove a release when it hasn't been installed?

File: .github/workflows/dagster.yml [hunk 2]
```diff
@@ -85,3 +111,72 @@ jobs:
...
+      - if: ${{ github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'dev-env') }}
+        name: Deploy dev environment
+        uses: glopezep/helm@v1.7.1 # Should be replaced with deliverybot/helm@v1 when https://github.com/deliverybot/helm/issues/66 is fixed
+        with:
+          helm: helm3
+          repository: "https://dagster-io.github.io/helm"
+          chart: 'dagster'
+          release: "${{ format('{0}-pr-{1}', inputs.image_name, github.event.number) }}"
+          namespace: 'dagster'
+          chart_version: '0.14.12'
```
Comment (chanange) [inline .github/workflows/dagster.yml:127]: Minor, we'll need to remember to update this over time to match staging/prod

**Review by chanange — APPROVED** [review]
Super cool feature! 🚀

File: .github/workflows/dagster.yml [hunk 3]
```diff
@@ -85,3 +111,72 @@ jobs:
...
+              nodeSelector:
+                cloud.google.com/gke-nodepool: core
+              runMonitoring:
+                enabled: true
+                startTimeoutSeconds: 300
+            dagit:
+              nodeSelector:
+                cloud.google.com/gke-nodepool: core
+            global:
+              postgresqlSecretName: "${{ format('pr-{0}', github.event.number) }}-psql-secret"
```
Comment (chanange) [inline .github/workflows/dagster.yml:139]: Might be better to include the repo name for this secret and the ssl secret (in the unlikely case where 2 repos have the same PR #)

```
pr-155-psql-secret                        Opaque                                1      2d5h
pr-155-ssl-secret                         kubernetes.io/tls                     2      2d4h
pr-4-ssl-secret                           kubernetes.io/tls                     2      7d2h
```
Also seems like the SSL secret for pr-4 wasn't cleaned up

Comment (chanange) [reply .github/workflows/dagster.yml:139]: Potential fix: https://github.com/cert-manager/cert-manager/issues/2993#issuecomment-651113618

Comment (AndreaGiardini) [reply .github/workflows/dagster.yml:139]: Fixed by https://github.com/20treeAI/infra/pull/609

File: .github/workflows/dagster.yml [hunk 4]
```diff
@@ -85,3 +111,72 @@ jobs:
...
+                nginx.ingress.kubernetes.io/auth-snippet: |
+                  auth_request_set $auth_resp_jwt $upstream_http_x_vouch_jwt;
+                  auth_request_set $auth_resp_err $upstream_http_x_vouch_err;
+                  auth_request_set $auth_resp_failcount $upstream_http_x_vouch_failcount;
+              dagit:
+                host: "${{ format('{0}-pr-{1}', inputs.image_name, github.event.number) }}.stage-elm.overstory.ai"
+                path: "/"
+                tls:
+                  enabled: true
+                  secretName: "${{ format('pr-{0}', github.event.number) }}-ssl-secret"
```
Comment (chanange) [inline .github/workflows/dagster.yml:175]:
```suggestion
                  secretName: "${{ format('{0}-pr-{1}', inputs.image_name, github.event.number) }}-ssl-secret"
```

**Pull request merged**